File: js/islandPerimeter.js

function islandPerimeter(grid){
    let perimeter = 0

    for (let i = 0; i < grid.length; ++i){
        for (let j = 0; j < grid[0].length; ++j){
            if (grid[i][j] == 1){
                // left
                if (j == 0 || j > 0 && grid[i][j-1] == 0){ // need || it's a wall
                    perimeter += 1
                }
                // right
                if (j == grid[i].length - 1 || j < grid[i].length - 1 && grid[i][j+1] == 0){
                    perimeter += 1
                }
                // bottom?
                if (i == grid.length - 1 || i < grid.length - 1 && grid[i+1][j] == 0){
                    perimeter += 1
                }
                // if i == len(grid) - 1 or i < len(grid) - 1 and grid[i+1][j] == 0:
                // top
                if (i == 0 || i > 0 && grid[i-1][j] == 0){
                    perimeter += 1
                }
                // how many sides are open? // f|| each side, add 1 to perimeter
            }
        }
    
    }
    return perimeter
}

const result = islandPerimeter([[1], [0]]);

File: py/islandPerimeter.py
def islandPerimeter(grid):
    perimeter = 0
    for i, line in enumerate(grid):
        for j, block in enumerate(line):
            if block == 1:
                # left
                if j == 0 or j > 0 and line[j-1] == 0: # need OR it's a wall
                    perimeter += 1
                # right
                if j == len(line) - 1 or j < len(line) and line[j+1] == 0:
                    perimeter += 1
                # bottom?
                if i == len(grid) - 1 or i < len(grid) - 1 and grid[i+1][j] == 0:
                    perimeter += 1
                # top
                if i == 0 or i > 0 and grid[i-1][j] == 0:
                    perimeter += 1

                # how many sides are open? # for each side, add 1 to perimeter

    return perimeter


result = islandPerimeter([[0,1,0,0],
                          [1,1,1,0],
                          [0,1,0,0],
                          [1,1,0,0]])